### PR TITLE
Adding Mod Settings integration and more natural message send timing

### PR DESCRIPTION
**Pull request by GoobyMain**
I have no idea how to use github, I hope this works, sorry if it doesn't or it's messy
Also sorry the script code is messy, I also have no idea how to mod TF2 and generally I'm not super great at coding, but at least it works lmao

I unnecessarily overcomplicated the mod to work more like how I wanted, figured some other people might like the changes so I might as well send it since I'd already made them

Changes:
- added a beginning of game message (ex. "glhf")
- added a configurable delay, for how long to wait until sending the message, because it bothered me in the original mod how the message would send immediately when the game ends, before everyone else
- added integration with the Mod Settings mod, which autoGG is now dependent on, to change ConVars instead of using the .cfg file; with the ConVars all at their defaults, the mod behaves the same as before
- changed the end of game message to send (after the configurable delay) differently depending on whether or not the game has an epilogue:
    - if there is no epilogue, sends immediately
    - if there is an epilogue, waits to send until you die
    - if there is an epilogue and you don't die, sends immediately -- without waiting for the delay -- after the epilogue ends and you lose control of you player
  these seemed like the most natural points when I would set down my controller and pull out my keyboard to type "gg" in chat

I also included a .zip file package for thunderstore that includes a manifest.json with the proper dependency listed for the Mod Settings mod, the package needs a license so I copied the license from this github, you probably want to take a look at it to make sure it's fine

Also not documented anywhere else, there's a bug where the "gg" message could send twice if the player died at the right time at the end of the epilogue, I have no idea if this would happen often enough to be worth fixing

### CODE REVIEW

**Comment by catornot** [comment]
I abandoned this repo, I guess some people still like these mods lol

btw cool stuff

File: auto-gg-thunderstore-package.zip
Comment (catornot) [inline auto-gg-thunderstore-package.zip:1]: this should be removed and the manifest, read me, etc should be moved to to the mod folder like [here](https://github.com/catornot/cat_or_not.Furnace)

I am using [r2template](https://github.com/catornot/r2template) to create and pack mods which uses that format

**Review by catornot — CHANGES_REQUESTED** [review]

**Comment by catornot** [comment]
actually since I don't plan on maintaining any of these mods you can just upload this forked version yourself to thunderstore ;)

File: auto-gg-thunderstore-package.zip
Comment (GoobyMain) [reply auto-gg-thunderstore-package.zip:1]: Cool! Thanks for the help :)

**Comment by catornot** [comment]
what 2 reviews showed up lol

**Comment by GoobyMain** [comment]
> what 2 reviews showed up lol

Might have been me? I still have no idea what's going on with github lol, sorry if it was me

**Pull request closed without merging**